pact/concrete-policies/fungible-quote-policy/fungible-quote-policy-v1.pact

```diff
@@ -159,8 +238,122 @@
+      sale-id:string )      
+
+      (with-read quotes sale-id { 
+        'id:= qtoken, 
+        'spec:= spec:object{quote-spec} 
+        'reserved:= reserved
+      }
+        (enforce (= qtoken token-id) "incorrect sale token")
+
+        (bind spec
+          { 'fungible := fungible:module{fungible-v2}
+            ,'amount := quote-amount:decimal
+          }
+          (let ((bid-id:string (get-bid-id sale-id buyer)))
+
+          (enforce (= quote-amount amount) "Bid can only be placed for full quote amount")
+
+          ;; Transfer amount from buyer to escrow-account
+          (fungible::transfer-create buyer (bid-escrow-account bid-id) (create-capability-guard (BID_PRIVATE bid-id)) (* amount price))
+
+          ;; Store bid
+          (write bids bid-id {
+            "token-id": token-id
+            ,"buyer": buyer
+            ,"buyer-guard": buyer-guard
+            ,"amount": amount
+            ,"price": price
+            ,"status": BID-STATUS-OPEN
+          })
+
+          (emit-event (BID bid-id sale-id token-id amount price buyer))
+        )))
+  )
+
+  (defun withdraw-bid:bool (bid-id:string sale-id:string)
+    (with-read bids bid-id
+      { 'token-id:= token-id,
+        'buyer:= buyer,
+        'buyer-guard:= buyer-guard,
+        'amount:= amount,
+        'price:= price,
+        'status:= status
+      }
+      (with-capability (BUYER buyer-guard)
+        (enforce (= status BID-STATUS-OPEN) "Bid is not open")
+        (with-read quotes sale-id { 'spec:= spec:object{quote-spec} }
+          (let* (
+            (fungible:module{fungible-v2} (at 'fungible spec)))
+
+            (with-capability (BID_PRIVATE bid-id)
+              (install-capability (fungible::TRANSFER (bid-escrow-account bid-id) buyer (* amount price)))
+              (fungible::transfer (bid-escrow-account bid-id) buyer (* amount price))
+            )
+            (update bids bid-id { 'status: BID-STATUS-WITHDRAWN })
+          )
+        )
+        (emit-event (BID-WITHDRAWN bid-id sale-id token-id amount price buyer)))
+    )
+  )
+
+  (defun accept-bid:bool (bid-id:string sale-id:string)
+    (with-read bids bid-id
+      { 'token-id:= token-id,
+        'buyer:= buyer,
+        'buyer-guard:= buyer-guard,
+        'amount:= amount,
+        'price:= price,
+        'status:= status
+      }
+      (enforce (= status BID-STATUS-OPEN) "Bid is not open")
+      (with-read quotes sale-id { 'spec:= spec:object{quote-spec} }
+        (let* (
+          (fungible:module{fungible-v2} (at 'fungible spec))
+          (seller-guard:guard (at 'seller-guard spec)))
+
+          (with-capability (SELLER seller-guard)
+            (update quotes sale-id { 'spec: { 'fungible: fungible, 'price: price, 'seller-guard: seller-guard, 'amount: amount }, 'reserved: buyer })
+            (update bids bid-id { 'status: BID-STATUS-ACCEPTED })
+          )
+        )
+      )
+      (emit-event (BID-ACCEPTED bid-id sale-id token-id amount price buyer))
```

Comment:
Let's add `QUOTE` event here, as we're also updating the quote

Reply:
Done.